torch/_inductor/ir.py

```diff
@@ -3777,9 +3775,35 @@ def __init__(self, count: int, device: torch.device):
 
 
 class ExternKernelAlloc(ExternKernel):
+    # Generate abi-compatible kernel names for shim kernels.
+    # Each individual shim kernel may have its own versioning rule.
+    # However, we don't expect we would end up with too many of such rules.
+    def _get_abi_compatible_kernel(self):
+        if not V.graph.cpp_wrapper:
+            return self.kernel
+
+        def sdpa_ver_fn():
+            # For sdpa, we need the v2 version only if any optional
+            # kwarg is missing.
```

Comment:
I thought the idea was to make all new models call the v2 version, so that we could delete the old one entirely? Doesn't using the v2 version only if an optional arg is missing make migration harder?

Reply:
> I thought the idea was to make all new models call the v2 version, so that we could delete the old one entirely? Doesn't using the v2 version only if an optional arg is missing make migration harder?

It's not about new models v.s. old models. Given the same model, the optional scale could be either None or have a default value, for different backends (e.g. cpu and cuda in our case). Regarding migrating to the V2 version, my understanding is that once the new V2 version becomes available on the serving side, we could always generate the V2 version. At that point, we may safely remove the old one.

Reply:
Ohh right. We want to keep generating calls to the old (v1) version where possible for now so the CUDA models keep working with old libtorch deployments. Got it